### ISSUE TITLE
Bug Report: Unexpected CreateMessageAsync Method Failure regarding Content-Type UTF-8

### ISSUE THREAD

**Issue by disintegr8te** [RageAgainstThePixel/OpenAI-DotNet#236]
# Bug Report

## Overview
The  `CreateMessageAsync`  method stopped working today, resulting in a BadRequest error. This issue is likely due to API changes made by OpenAI, as the error message indicates a problem with the  `Content-Type`  header being set to  `'application/json; charset=utf-8'`  instead of the expected  `'application/json'`.

## To Reproduce
1.  Initialize the API client with necessary configurations.
2.  Attempt to create a message using  `CreateMessageAsync`  method by providing a valid  `threadId`  and  `CreateMessageRequest`  object.
3.  Observe the BadRequest error with the response body indicating an  `Unsupported content type`  error.

`An error occurred: CreateMessageAsync Failed! HTTP status code: BadRequest | Response body: { "error": { "message": "Unsupported content type: 'application/json; charset=utf-8'. This API method only accepts 'application/json' requests, but you specified the header 'Content-Type: application/json; charset=utf-8'. Please try again with a supported content type.", "type": "invalid_request_error", "param": null, "code": "unsupported_content_type" } } Help me fix somehow`

## Expected behavior
The expected behavior was that the  `CreateMessageAsync`  method would successfully post the message data to the corresponding API endpoint without encountering a BadRequest error. The message creation should be processed by the OpenAI API and return a success response.

## Screenshots
_Not applicable, as the error is within the response body of an API call._

## Additional context

-   The method was working as expected until yesterday, suggesting a change in the API requirements regarding HTTP headers, specifically the  `Content-Type`.
-   The response error explicitly states that the  `Unsupported content type: 'application/json; charset=utf-8'`  is not accepted, hinting at a recent change in OpenAI's API to enforce a strict  `'application/json'`  content type without charset specification.
-   A potential resolution would involve adjusting the HTTP request made by the  `CreateMessageAsync`  method to strictly set the  `Content-Type`  header to  `'application/json'`  without including the charset.

## Suggested Solution
_I suggest adjusting the handling of the  `Content-Type`  header within the request preparation of the  `CreateMessageAsync`  method to comply with the API requirements.

**Comment by StephenHodgson** [comment]
I have not seen any changes made to the spec for a few days

https://github.com/openai/openai-openapi/pulls

**Comment by disintegr8te** [comment]
Yes, dont see a Change there as well. Have this behaviour since Yesterday in an untouched Code.
But a working Workaround is to Override it in the httpClient to "application/json" monly.

**Comment by StephenHodgson** [comment]
Interesting change from the node client:

https://github.com/openai/openai-node/pull/664/files#diff-033ecd338eec0994fe53c0896219baee901e20068b8fbe6259a23d5759584b06R66

**Comment by kalleandreasson** [comment]
I have the exact same issue when I'm trying to create a new assistant

**Comment by StephenHodgson** [comment]
Sorry but I am unable to repro. Ran all the tests just now

![image](https://github.com/RageAgainstThePixel/OpenAI-DotNet/assets/13334553/992f8848-5f2f-4e0a-9fa0-c6c140ce7750)

**Comment by shabalinskyartem** [comment]
Hi guys, I have the same issue like a @disintegr8te

**Comment by StephenHodgson** [comment]
Any updates here?
